crates/net/network/src/transactions/fetcher.rs

```diff
@@ -1072,13 +1081,45 @@ impl Default for TransactionFetcher {
     }
 }
 
+#[derive(Debug, Clone)]
```

Comment:
```suggestion
#[derive(Debug, Clone, Constructor)]
```

one last thing to address, asides that which @onbjerg says, is to import `derive_more::Constructor`, then replace all the places in the code where you make an instance of the type. so this
```rust
TxSizeMetadat::new(peer_id, 0)
```
instead of making it like this 
```rust
TxSizeMetadata {
    peer_id: <value>,
    len: <value>,
}
```